src/Compiler/TypedTree/TypedTreePickle.fs

```diff
@@ -2252,7 +2258,7 @@ and u_attribkind st =
 
 and u_attrib st : Attrib =
     let a, b, c, d, e, f = u_tup6 u_tcref u_attribkind (u_list u_attrib_expr) (u_list u_attrib_arg) u_bool u_dummy_range st
-    Attrib(a, b, c, d, e, None, f)  // AttributeTargets are not preserved
+    Attrib(a, b, [], c, d, e, None, f)  // AttributeTargets are not preserved (u_list u_ty)
```

Comment:
What will happen if an older compiler sees metadata containing type arguments?

Reply:
If I understand your question correctly, this relates to a situation where in an assembly that actually uses generic attributes (i.e. actually applying attributes to "things") is referenced by another assembly being built by an older version of fsc. In that case, I think the type arguments of used attributes will need to be digested into the "TyConRef" field of the typed Attrib; i.e., it would need to become a new type definition?

Reply:
Sorry, I've read this change wrong. Not keeping the data is sad, but at the same time changing the metadata format is probably a no-go here for now.

Reply:
Would that kind of change just be another RFC/PR, or are there other complexities that need to be dealt with?

Reply:
The problem is we try to maintain the metadata compatibility with the older compilers, and currently there's no preferred way to add new data to it without breaking existing compilers. @vzarytovskii Has there been any changes in this direction?

Reply:
The precedence we used to take before:

- Absolutely maintain backwards compatibility for existing source code (i.e. code not using the feature)
- Have a compiler error if this is used in FSharp.Core
- New code using this feature will force consumers of such code to upgrade their tooling (IDE,SDK). 

There is an alternative approach of not breaking the pickle format and instead loading the needed info from .NET IL metadata ;; but I do not think this path has been taken before (even though it would be safer from a compatibility perspective, because we do not need to know how to load it for interop scenarios anyway)

Reply:
I would assume that the purpose of the pickled resources is for the sake of performance improvements (correct me if I'm wrong), in which case I would think that there are some kind of benchmarks concerning the gain of using pickling as opposed to loading the same resources from the IL? Assuming that the cost in this case wouldn't be great, and that it is feasible, loading the info from the IL and not breaking backward compatibility sounds like the better option.